### PR TITLE
Open agent + view on agent builder mode for empty project

### PR DESCRIPTION
**Pull request by kaumini**
- [ ] Added reusable UI components to the ui-toolkit. Follow the [instructions](../submodules/wso2-vscode-extensions/workspaces/common-libs/ui-toolkit/README.md) when adding the component.
- [ ] Use ui-toolkit components wherever possible. Run `npm run storybook` from `submodules/wso2-vscode-extensions/workspaces/common-libs/ui-toolkit` to view current components.
- [ ] Matches with the native VSCode look and feel.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

 

## Summary

- Enable agent-builder mode for empty projects.
- Open the `AddAgent` view when one eligible empty package is open.
- Prevent fallback navigation when the agent-builder landing opens successfully.
- Add agent-builder mode detection to configuration and state-machine APIs.
- Add tests for empty and populated integrations.
- Add a VS Code launch configuration for agent-builder mode.

### CODE REVIEW

File: packages/ballerina-extension/src/utils/config.ts [hunk 1]
```diff
@@ -246,6 +246,11 @@ export function isInDevant(): boolean {
     return !!process.env.CLOUD_STS_TOKEN;
 }
 
+export function isAgentBuilderMode(): boolean {
```
Comment (gigara) [inline packages/ballerina-extension/src/utils/config.ts]: Shall we use the same logic on the Integrator side? 
Ref:  https://github.com/wso2/product-integrator/blob/agent-builder/wi/wi-extension/src/productMode.ts#L30

Comment (kaumini) [reply packages/ballerina-extension/src/utils/config.ts]: Resolved with [16ffc97](https://github.com/ballerina-platform/ballerina-vscode/pull/834/commits/16ffc972e01fbafa7bf9bb99c95d5a4b11d86bb7)

File: packages/ballerina-extension/src/stateMachine.ts [hunk 2]
```diff
@@ -972,6 +974,7 @@ export const StateMachine = {
         const state = stateService.getSnapshot().value;
         return typeof state === 'object' && 'viewActive' in state && state.viewActive === "viewReady";
     },
+    productMode: () => { return stateService.getSnapshot().context.productMode; },
```
[the remark this answers is not in the document]
Comment (kaumini) [reply packages/ballerina-extension/src/stateMachine.ts:977]: StateMachine.isAgentBuilderMode() is not part of the published extension API. StateMachine is an internal module that is neither returned from activate() nor re-exported by the webpack entry point, and src/ is excluded from the VSIX. A repository-wide search found only two consumers, both internal to this extension, and both were migrated to productMode(). Additionally, the old accessor was introduced on this feature branch and has not been released from the base branch. Therefore, no compatibility adapter is required.

**Review by gigara — APPROVED** [review]
LGTM
